scripts/zones/Windurst_Waters/npcs/Leepe-Hoppe.lua

```diff
@@ -49,7 +49,11 @@ function onTrigger(player,npc)
         if (player:hasKeyItem(dsp.ki.MOON_BAUBLE)) then -- Default text after acquiring moon bauble and before fighting Fenrir
             player:startEvent(845,0,1125,334);
         elseif (player:hasKeyItem(dsp.ki.WHISPER_OF_THE_MOON)) then -- First turn-in
-            player:startEvent(846,0,13399,1208,1125,0,18165,13572);
+            local availRewards = 0
+            if (not player:hasKeyItem(dsp.ki.TRAINERS_WHISTLE)) then availRewards = availRewards + 128; end -- Mount Pact
+-- TODO: add logic to determine if player has already / or can obtain Mount to remove it from list of rewards
```

Comment:
Isn't the TODO satisfied with checking for having Fenrir's Whistle?

```lua
and not player:hasKeyItem(dsp.ki.FENRIR_WHISTLE)
```

Reply:
No, this doesn't satisfy that condition because you should also not recieve the mount option if you already own the mount (as well as) not having done the quest

Reply:
oh wait... i thought you meant TRAINERS_WHISTLE... I don't know about fenrir's whistle, I have not done anything with mount quests... I've only worked with this quest and know what the params are.

Reply:
Currently, the list of mounts you're eligible to use is built on the list of key items you possess. (As of right now, key items 3072 through 3092)

In the case of obtaining the Fenrir, you directly earn `dsp.ki.FENRIR_WHISTLE` from selecting the mount option as your reward. Other mounts there is an item you trade to Mapitoto and receive the key item (eg: login rewards, red crab you obtained the item 10052, traded to Mapitoto and receive key item 3076)

Reply:
Mounts are fully functional outside the actual quest to get the Trainer's Whistle (Full Speed Ahead!), actually checking if you have a Trainer's Whistle (if you have a companion key item, you just have access to them), and the turn in items/key items being rewarded in appropriate places.

Reply:
Well, if that's the case... then this'll do it...